### DISCUSSION TITLE
Custom filter for data inside a DateTree

### DISCUSSION THREAD

**Discussion by isbincApps** [diatche/gun-util#19] [Q&A] [answered]
the library works great, I would like to know how to search for some field in DateTree, something like.
`tree.map (x => x.name.indexOf (`abc`)> -1)`

**Comment by diatche** [comment, marked as answer]
Hi there!

`DateTree` allows working with multiple Gun references like they are a single Gun reference, but not all chain methods are available. Currently, only `get` and `on` are available.

To filter data inside of a date tree inside of a date range, you can do the following:

```javascript
tree.on((data, date) => console.log(data), { gte: '2020-01-01', lt: new Date() });
```

At the moment, if you want to filter the data inside a date tree by other criteria (other than by date), you need to apply your own logic on the data you receive inside the callback.

**Comment by isbincApps** [comment]
ok @diatche thanks, It would be great if you could implement something like queries in the library

**Comment by isbincApps** [comment]
Hi @diatche , how can i build a pagination requests in the datatree?

**Comment by diatche** [comment]
> Hi @diatche , how can i build a pagination requests in the datatree?

Hey @isbincApps ,

You could specify the date range for each page. Here's a simple example:

```javascript
const msInDay = 86400000;

let dateTree = new dateTree(/* root ref */);
let pageDataSub = null;
let startDate = new Date(2020, 10, 10);

// Each page shows data for one day
const subscribeToPageData = (page, startDate, callback) => {
    if (pageDataSub) {
        // Unsubsribe from previous page
        pageDataSub.off();
    }
    let pageDateStart = page * msInDay + startDate.valueOf();
    let pageDateEnd = pageDateStart + msInDay;
    pageDataSub = dateTree.on(callback, {
        gte: pageDateStart,
        lt: pageDateEnd,
    });
    return pageDataSub;
}

//...
// When page 1 requested:
subscribeToPageData(1, startDate, data => {
    // Do something with data on page 1
});


//...
// When page 2 requested:
subscribeToPageData(2, startDate, data => {
    // Do something with data on page 2
});
```

There are other approaches, this is only one of them, but this should do the trick.

**Comment by isbincApps** [comment]
that's great @diatche , thanks
